ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/ks/KsLocator.java

```diff
@@ -108,20 +146,42 @@
 
       keysByPartition.putIfAbsent(metadata.partition(), new LinkedHashSet<>());
       keysByPartition.get(metadata.partition()).add(key);
+      metadataList.add(metadata);
+    }
+    return new Metadata(metadataList, Optional.of(keysByPartition));
+  }
 
-      if (locationsByPartition.containsKey(metadata.partition())) {
-        continue;
-      }
+  private Metadata getMetadataForAllPartitions(final Set<Integer> filterPartitions) {
```

Comment:
We should add tests in `KsLocatorTest` to make sure that the data structures are populated correctly with all active and standby partitions the underlying store has